### PR TITLE
Update haxe-dev and support release candidate in checkver Updated hash for hashlink

### PR DESCRIPTION
**Pull request by mundusnine**

### CODE REVIEW

File: bucket/haxe-dev.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     ],
     "checkver": {
         "url": "https://github.com/HaxeFoundation/haxe/releases",
-        "re": "([\\d.]+-preview[\\d.]+)"
+        "re": "(\\d\\.\\d\\.\\d-preview\\.\\d|\\d\\.\\d\\.\\d\\-rc\\.\\d)"
```
Comment (Ash258) [inline bucket/haxe-dev.json]: Why don't go with simple regex?

`([\d\.]+)-(rc|preview)[\d\.]+`

Comment (mundusnine) [reply bucket/haxe-dev.json]: I just started doing regex and I am going with what I can find and I didn't know we could do this so simply. Will change it.

Comment (mundusnine) [reply bucket/haxe-dev.json]: I tried your command and it doesn't work

Comment (Ash258) [reply bucket/haxe-dev.json]: Proof

Comment (mundusnine) [reply bucket/haxe-dev.json]: The one in the pictures isn't the same as the one in your comment. But I will change it to what is in your picture. 
image

Comment (Ash258) [reply bucket/haxe-dev.json]: It's same. It's only escaped to confort json.

**Pull request merged**